cli/src/main/java/com/devonfw/tools/ide/process/ProcessContextImpl.java

```diff
@@ -313,22 +315,28 @@ private String addExecutable(String exec, List<String> args) {
   private void performLogOnError(ProcessResult result, int exitCode, String interpreter) {
 
     if (!result.isSuccessful() && (this.errorHandling != ProcessErrorHandling.NONE)) {
-      String message = createCommandMessage(interpreter, " failed with exit code " + exitCode + "!");
-      if (this.errorHandling == ProcessErrorHandling.THROW_CLI) {
-        throw new CliProcessException(message, result);
-      } else if (this.errorHandling == ProcessErrorHandling.THROW_ERR) {
-        throw new IllegalStateException(message);
-      }
-      IdeSubLogger level;
+      IdeLogLevel ideLogLevel;
+      String message = createCommandMessage(interpreter, "\nfailed with exit code " + exitCode + "!");
+
       if (this.errorHandling == ProcessErrorHandling.LOG_ERROR) {
-        level = this.context.error();
+        ideLogLevel = IdeLogLevel.ERROR;
+        context.error(message);
```

Comment:
IMHO the log-level is the only code dependent on the `ProcessErrorHandling` in order to control the log-level at which the message is logged after the `if` block.
Why do you still log the message manually in each condition block?
```
this.context.level(ideLogLevel).log(message);
```
BTW: The enum `ProcessErrorHandling` could define an abstract method `getLogLevel()` so that I do not need the `if ... else if ...` logic here and when creating a new enum constant, I get a compile error unless I implement `getLogLevel()`, so I do not accidentally end up in the `else` block with probably undesired behaviour.